### DISCUSSION TITLE
Is there a django style model manager?

### DISCUSSION THREAD

**Discussion by simkimsia** [collerek/ormar#426] [Q&A] [answered]
i like how your docs cover python and django style for queries. https://collerek.github.io/ormar/queries/filter-and-sort/#django-style_5

Well done

Is there a way to have django style model managers using ormar?

Really happy to discover ormar. Thank you

**Comment by collerek** [comment, marked as answer]
What do you mean by django model managers? 

You have "manager" (`QuerySet`) under `objects` like in django. 
If you want to register a new one you can manually create a `QuerySet` and assign it to the class, or subclass `QuerySet` and add your own methods.

**Reply by simkimsia** [reply]
Yes that's what i mean by "manager" 

At present, sometimes i have multiple "managers" for the same model.

Like Book.creators, Book.updators, Book.readers, etc 

So I suppose I can define them as per normal like in Django?

**Reply by collerek** [reply]
I didn't test approach like this, but in general yes - it should work. Under the hood, it's a QuerySet that refers to the Model class and you can create one (or more) yourself.

**Reply by simkimsia** [reply]
thank you . i will try it out and document what works. thanks for writing ormar

**Comment by BlackLIonz** [comment]
The only way I found, to add custom functions to a 'Queryset'. You need to override `objects` method in MetaModel and add it to your model:
``` python
class NoteQueryset(QuerySet):
    async def get_or_404(self, *args: Any, **kwargs: Any) -> Optional['Note']:
        try:
            return await self.get(*args, **kwargs)
        except ormar.NoMatch:
            raise HTTPException(status_code=404, detail='Note not found')


class NoteModelMetaclass(ModelMetaclass):
    @property
    def objects(cls: Type["T"]) -> "QuerySet[T]":  # type: ignore
        _ = super().objects
        return NoteQueryset(model_cls=cls)


class Note(Base, metaclass=NoteModelMetaclass):
```
If you know the better way, pls share it
